### PR TITLE
not reuse docker terminal

### PR DESCRIPTION
**Pull request by yungez**

### CODE REVIEW

**Comment by yungez** [comment]
@zikalino  docker terminal is not reused because each time user run playbook docker cmd might be different. local ansible terminal is reusable since it's just a normal console.

**Comment by zikalino** [comment]
@yungez so the old terminal is destroyed and a new terminal is created, right?
what could be typical differences in docker parameters?

**Comment by yungez** [comment]
old terminal will keep at there for user to check their previous input/output. typical changes in commands like credentials changed, telemetry config changed, playbook file changed.

**Comment by yaweiw** [comment]
@yungez do you consider to limit num of terminal windows opened in the extension?

File: src/terminalExecutor.ts [hunk 1]
```diff
@@ -21,9 +24,19 @@ export class TerminalExecutor {
         reuseTerminal: boolean,
         cb: Function): void {
         if (!reuseTerminal || (this.terminals === undefined || this.terminals[terminalName] === undefined)) {
+            if (!terminalCount[terminalName]) {
+                terminalCount[terminalName] = 0;
+            }
+
+            if (terminalCount[terminalName] >= MAX_TERMINAL_COUNT) {
+                vscode.window.showErrorMessage('Reached max count of terminal: ' + terminalName + ', please delete unused terminals.');
```
Comment (yaweiw) [inline src/terminalExecutor.ts]: Reached the max limit of active terminals ....

**Pull request merged**